client/components/clipboard-button-input/index.jsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import classnames from 'classnames';
+import omit from 'lodash/object/omit';
+
+/**
+ * Internal dependencies
+ */
+import ClipboardButton from 'components/forms/clipboard-button';
+import FormTextInput from 'components/forms/form-text-input';
+
+export default React.createClass( {
+	displayName: 'ClipboardButtonInput',
+
+	propTypes: {
+		value: PropTypes.string,
+		className: PropTypes.string
+	},
+
+	getInitialState() {
+		return {
+			isCopied: false
+		};
+	},
+
+	getDefaultProps() {
+		return {
+			value: ''
+		};
+	},
+
+	componentWillUnmount() {
+		clearTimeout( this.confirmationTimeout );
+		delete this.confirmationTimeout;
+	},
+
+	showConfirmation() {
+		this.setState( {
+			isCopied: true
+		} );
+
+		this.confirmationTimeout = setTimeout( () => {
+			this.setState( {
+				isCopied: false
+			} );
+		}, 4000 );
+	},
+
+	render() {
+		const { value, className } = this.props;
+		const classes = classnames( 'clipboard-button-input', className );
+
+		return (
+			<span className={ classes }>
```

Comment:
>if `value` is null here should this component just return `null` since there would be nothing to copy?

We provide a default value of an empty string here, so at the very least it should never be `null`. I'd think any `null`-checking variations should be handled by a parent component. I could see it being useful to add a `disabled` option to disable both the text input and button if, for example, the component is used in a form that is loading data asynchronously. In these cases, `disabled` seems preferable to the element disappearing and reappearing.

I'll see to adding a `disabled` prop :smiley: